### PR TITLE
Add news API

### PR DESCRIPTION
**Pull request by noramehesz**

### CODE REVIEW

File: lib/resources/datav2/rest_v2.ts [hunk 1]
```diff
@@ -62,30 +65,54 @@ export function dataV2HttpRequest(
...
 ): AsyncGenerator<any, void, unknown> {
   let pageToken: string | null = null;
-  let totalItems = 0;
-  const limit = options.limit;
+  let received = 0;
+  const pageLimit = options.pageLimit
+    ? Math.min(options.pageLimit, V2_MAX_LIMIT)
+    : V2_MAX_LIMIT;
+  delete options.pageLimit;
+  const totalLimit = options.limit ?? V2_MAX_LIMIT;
```
Comment (gnvk) [inline lib/resources/datav2/rest_v2.ts:102]: Maybe I'm missing something here, but this doesn't seems right. If no `limit` is provided by the user, you set it to `V2_MAX_LIMIT`, which is a single page? That basically removes the pagination implementation provided by the SDK.

File: lib/resources/datav2/news_websocket.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    };
+    this.conn.send(this.msgpack.encode(unsubMsg));
+  }
+
+  updateSubscriptions(msg: { news: Array<string> }): void {
+    this.log(
+      `listening to streams:
+        news: ${msg.news}`
+    );
+  }
```
Comment (Mark-Pintye) [inline lib/resources/datav2/news_websocket.ts:60]: this only logs, no action

File: lib/resources/datav2/rest_v2.ts [hunk 3]
```diff
@@ -539,3 +566,78 @@ export async function getCryptoSnapshot(
...
+  const query = {} as any;
+  query.symbols =
+    options.symbols?.length > 0 ? options.symbols.join(",") : null;
+  query.start = options.start ? options.start : null;
+  query.end = options.end ? options.end : null;
+  query.sort = options.sort ? options.sort : null;
+  query.includeContent = options.includeContent ? options.includeContent : null;
+  query.excludeContentless = options.excludeContentless
+    ? options.excludeContentless
+    : null;
```
Comment (Mark-Pintye) [inline lib/resources/datav2/rest_v2.ts:605]: i dont think these are needed

Comment (Mark-Pintye) [reply lib/resources/datav2/rest_v2.ts:605]: for the lines above either

File: lib/resources/datav2/rest_v2.ts [hunk 4]
```diff
@@ -539,3 +566,78 @@ export async function getCryptoSnapshot(
...
+    if (limit < 1) {
+      break;
+    }
+
+    const resp: AxiosResponse<any> = await dataV2HttpRequest(
+      "/v1beta1/news",
+      { ...params, limit: limit, page_token: pageToken },
+      config
+    );
+    resp.data.news.map((n: RawAlpacaNews) => result.push(AlpacaNews(n)));
```
Comment (Mark-Pintye) [inline lib/resources/datav2/rest_v2.ts:635]: map is used as a pure function that has no sideeffect and the return value is used . For this usecase foreach is the better approach

File: lib/resources/datav2/news_websocket.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  onNews(fn: (n: AlpacaNews) => void): void {
+    this.on(EVENT.NEWS, (n: AlpacaNews) => fn(n));
+  }
+
+  dataHandler(data: Array<RawAlpacaNews>): void {
+    data.forEach((element: RawAlpacaNews) => {
+      if ("T" in element) {
```
Comment (Mark-Pintye) [inline lib/resources/datav2/news_websocket.ts:68]: condition not needed

File: lib/resources/datav2/rest_v2.ts (hunk 1 above)
Comment (noramehesz) [reply lib/resources/datav2/rest_v2.ts:102]: This expression means that we use `options.limit` for the `totalLimit` but if it's null then go with the max value.

Comment (gnvk) [reply lib/resources/datav2/rest_v2.ts:102]: So how does the user query all trades between start and end?

Comment (noramehesz) [reply lib/resources/datav2/rest_v2.ts:102]: No, the will query with `limit = 10000`.

**Review by Mark-Pintye — APPROVED** [review]

**Pull request merged**